### ISSUE TITLE
Uncaught ReferenceError: ref is not defined

### ISSUE THREAD

**Issue by RdeWilde** [Leniolabs/layoutit-grid#89]
I get this error when I try to load the component

```
import { createApp } from 'vue'
import LayoutEditor from 'layoutit-grid/src/components/LayoutEditor'
// ...
createApp(LayoutEditor).mount("#apppp")
```

![image](https://user-images.githubusercontent.com/184016/127329268-9712bade-79a7-4cb3-b915-d09ddc1c55c5.png)

Seems like the function is in vue/compat but my browser cannot resolve it.

I am using Laravel mix for the webpack configuration.

I tried to resovle vue to vue-compat, but not sure if that did work correctly:

```
mix.webpackConfig({
    resolve: {
        alias: {
            vue: '@vue/compat'
        }
    }
});
```

**Comment by RdeWilde** [comment]
Should this be on top of src/store.js?

`import { ref, watch } from '@vue/compat'`

**Comment by patak-js** [comment]
Using the app as a library is not in the scope of the project at this point. You are using Webpack and the app is built with Vite for example. Would you tell us more about your use case? I'm curious 👀

**Comment by patak-js** [comment]
Hey @RdeWilde, I see that you have published it in npm https://www.npmjs.com/package/layoutit-grid
It is not good if you clone and publish an app like this to npm, as it looks like it is your own creation in npm. I would like to ask you to donate the layoutit-grid package name to us, you can invite the username patak there and I can deal with it. Thanks

**Comment by RdeWilde** [comment]
I am trying to use it in our CMS so that clients can compose their photogrid for their website. How should I continue to fix this?

I published it because it is the only way to use it in NPM. Glad to donate, where can i do this? 

Hopefully you can help.

**Comment by RdeWilde** [comment]
@patak-js I invited you at NPM

**Comment by patak-js** [comment]
Thanks @RdeWilde. To give us the npm package name, you have to go to https://www.npmjs.com/package/layoutit-grid/access and invite the user `patak`, then you can remove yourself or I'll do it so the package is from then on managed by Leniolabs.

About your use case, interesting! At this point, I think the only way to consume the tool in other applications is by setting up an iframe and point it to grid.layoutit.com. If this works, then it is the easiest path forward. If you want to only use certain pieces, or start modifying how it works, the tool is not currently apt for that. We have discussed in the past to extract some parts of it but there are a few things to do first (like starting using a state management lib and create a good external API for the components).
You are free to fork the code though and modify it for your application as the license allows it. You should add a banner in the readme stating the reason for the fork for example, and you could publish parts of it to some other package name. But it will be kind of hard. I think that it may be a better idea to grab the code maintaining the license and use it directly in your project. If you manage to get a good setup for the library mode though, you could also contribute it back to this repo so others can use it.

**Comment by patak-js** [comment]
> @patak-js I invited you at NPM

Oh, thanks!

**Comment by patak-js** [comment]
I published a new version 2.1.0 to the package with a note about the current state. Thanks again @RdeWilde. Let us know if you end up reworking what we currently have for your use case. If we end up re-packaging the tool as a lib (it is a possibility that we discussed, but other priorities like adding support for flex are in our way), we'll let you know

**Comment by RdeWilde** [comment]
Awesome @patak-js will look into it. If you have any guidance om how to approach it, feel free to share and I'll try to take it into account.
